File: src/main/scala/croissong/starter/PostRouter.scala
package croissong.starter

import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.Configuration
import org.springframework.http.MediaType
import org.springframework.web.reactive.function.server.RequestPredicates
import org.springframework.web.reactive.function.server.RouterFunction
import org.springframework.web.reactive.function.server.RouterFunctions
import org.springframework.web.reactive.function.server.ServerResponse


@Configuration class PostRouter {
  @Bean def route(greetingHandler: PostHandler): RouterFunction[ServerResponse] =
    RouterFunctions.route(RequestPredicates.GET("/rest/post/1").and(RequestPredicates.accept(MediaType.TEXT_PLAIN)), greetingHandler.hello)
}

File: src/main/java/croissong/starter/PostRouter.java
package croissong.starter;

import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Configuration;
import org.springframework.http.MediaType;
import org.springframework.web.reactive.function.server.RequestPredicates;
import org.springframework.web.reactive.function.server.RouterFunction;
import org.springframework.web.reactive.function.server.RouterFunctions;
import org.springframework.web.reactive.function.server.ServerResponse;

@Configuration
public class PostRouter {

    @Bean
    public RouterFunction<ServerResponse> route(PostHandler greetingHandler) {

        return RouterFunctions
                .route(RequestPredicates.GET("/rest/post/1").and(RequestPredicates.accept(MediaType.TEXT_PLAIN)), greetingHandler::hello);
    }
}